katcp/kattypes.py

```diff
@@ -235,10 +235,10 @@ class Bool(KatcpType):
     encode = lambda self, value, major: value and b"1" or b"0"
 
     def decode(self, value, major):
-        if value not in (b"0", b"1"):
+        if value not in (b"0", b"1", "0", "1"):
             raise ValueError("Boolean value must be b'0' or b'1' but is '%s'."
                              % (value,))
-        return value == b"1"
+        return value == b"1" or value == "1"
```

Comment:
It's been a long time since I used this package so maybe I'm missing some context about how kattypes is used, but this seems a bit odd. I can understand decoded strings being either bytes or str because the katcp protocol has only a single type for text and raw data. But encoding should always produce bytes (because that's what sockets work on) and hence decoding should always expect bytes, not well-maybe-its-bytes-or-maybe-its-not. It smells like the calling code isn't  keeping a clean separation between encoded data on the wire and decoded text.

Reply:
@bmerry Your memory (and nose) is spot on.

The specific method that triggered this change was `katcp.Sensor.parse_params`, which uses the sensor's underlying type to decode the parameters.  In this context, the word `formatted` in the variable name indicates that it should be "formatted" like it would be serialised on the wire (i.e., byte strings).
https://github.com/ska-sa/katcp-python/blob/df7f3a425c6387b9b3c125d5db7c37fdf748a472/katcp/core.py#L1568

When we use it from the inspecting client, we pass a `sensor_info` object that will have native strings instead of byte strings.  The change to native strings was part of the previous round of migration to Python 3.
https://github.com/ska-sa/katcp-python/blob/df7f3a425c6387b9b3c125d5db7c37fdf748a472/katcp/inspecting_client.py#L872-L874

Yes, we could fix the usages in inspecting client (and fake inspecting client) instead.  I'm not sure if there are other bad usages of `decode`, but I imagine they will show up easily enough.

Reply:
@bmerry and @ajoubertza 
I addressed the issues with parsing parameters for boolean sensors where `parse_params` is called.
it is a bit hacky so any input if and when you are available would be appreciated.